### PR TITLE
[ONNX] Fix export of images for KeypointRCNN

### PR DESCRIPTION
**Pull request by KsenijaS**
Add support in KeypointRCNN for:
exporting for image w/ detection -> running on image w/ no detection
exporting for image w/ no detection -> running on image w/ detection

### CODE REVIEW

**Comment by KsenijaS** [comment]
@neginraoof please review

File: torchvision/models/detection/keypoint_rcnn.py [hunk 1]
```diff
@@ -253,10 +253,9 @@ def __init__(self, in_channels, num_keypoints):
 
     def forward(self, x):
         x = self.kps_score_lowres(x)
-        x = misc_nn_ops.interpolate(
-            x, scale_factor=float(self.up_scale), mode="bilinear", align_corners=False
+        return torch.nn.functional.interpolate(
```
Comment (fmassa) [inline torchvision/models/detection/keypoint_rcnn.py:256]: I've merged a PR that makes a conflict here, can you rebase?

File: torchvision/models/detection/keypoint_rcnn.py [hunk 2]
```diff
@@ -253,10 +253,9 @@ def __init__(self, in_channels, num_keypoints):
...
+            x, scale_factor=float(self.up_scale), mode="bilinear", align_corners=False, recompute_scale_factor=False
```
Comment (fmassa) [inline torchvision/models/detection/keypoint_rcnn.py:257]: Isn't `up_scale` a integer (equals to 2)? As such, is there a difference between `recompute_scale_factor=True` and `False`? Or is it because the `output_size` in this case gets registered as a constant?

Comment (neginraoof) [reply torchvision/models/detection/keypoint_rcnn.py:257]: With recompute_scale_factor, if input has a dim = 0, the calculated scale_factor will be 0. ONNX Runtime won't support this (throws an error if scale <= 0 in Resize op).

Just a note that I think the default value for recompute_scale_factor is changing to False soon.

Comment (fmassa) [reply torchvision/models/detection/keypoint_rcnn.py:257]: > With recompute_scale_factor, when input has a 0 dim, the inferred scale_factor will be 0.

This to me looks like a bug while inferring the scale_factor, as the batch dimension shouldn't affect the other sizes of the tensor (which are still valid).

Note that we use tensors of shape `[0, C, H ,W]`, so it is fully-possible to infer the correct scale factor I believe

File: torchvision/models/detection/keypoint_rcnn.py (hunk 1 above)
Comment (KsenijaS) [reply torchvision/models/detection/keypoint_rcnn.py:254]: I rebased it, conflict solved

**Comment by KsenijaS** [comment]
@fmassa can this PR be merged? Thanks!

**Comment by fmassa** [comment]
@KsenijaS I've made a comment on the top, not sure if you agree with it?

File: torchvision/models/detection/keypoint_rcnn.py (hunk 2 above)
Comment (neginraoof) [reply torchvision/models/detection/keypoint_rcnn.py:255]: I looked more into this. Looks like this is an error in inferring scale_factor in ONNX Runtime.
Upsample_bicubic2d is exported to ONNX and running in ONNX Runtime if you have an input like: torch.randn(1, 2, 4, 6, requires_grad=True)
ONNX graph also looks ok. 
But the same mode fails for an input like torch.randn(0, 2, 4, 6, requires_grad=True)
The reason is that ONNX Runtime is inferring scale for across all dimensions:
    for (size_t i = 0, end = input_dims.size(); i < end; ++i) {
      scales[i] = static_cast<float>(output_dims[i]) / static_cast<float>(input_dims[i]);
    }
For a 4D input, output_dims is inferred output shape (e.g. [n, c, h*scale, w*scale]) and input_dims is actual input shape (e.g. [n, c, h, w]).
This is a bug in such edge cases where n = 0.
We will create an issue for ORT to address this.

**Comment by neginraoof** [comment]
@fmassa Thanks a lot!
